vendor/engines/split_accounts/spec/models/split_account_spec.rb

```diff
@@ -111,4 +111,107 @@
+        end
+      end
+
+      it "suspends parent split_account" do
+        expect(split_account.suspended?).to eq(true)
+      end
+    end
+
+    context "when any subaccount is suspended" do
+      let(:suspended_subaccount) { build(:setup_account, suspended_at: Time.current) }
+
+      let(:split_account) do
+        build(:split_account, without_splits: true).tap do |split_account|
+          split_account.splits << build(:split, percent: 50, extra_penny: true, parent_split_account: split_account, subaccount: suspended_subaccount)
+          split_account.splits << build(:split, percent: 50, extra_penny: false, parent_split_account: split_account, subaccount: create(:setup_account))
+          split_account.save
+        end
+      end
+
+      it "suspends the parent split_account" do
+        expect(split_account.suspended?).to eq(true)
+      end
+
+      context "when subaccount unsuspends" do
+        before do
+          split_account
+          suspended_subaccount.suspended_at = nil
+          suspended_subaccount.save
+        end
+
+        it "keeps parent split_account suspended" do
+          expect(split_account.reload.suspended?).to eq(true)
+        end
+      end
+    end
+  end
+
+  describe "#expired?" do
+    context "when subaccounts and parent account are not expired" do
+      let(:split_account) do
+        build(:split_account, without_splits: true).tap do |split_account|
+          split_account.splits << build(:split, percent: 50, extra_penny: true, parent_split_account: split_account, subaccount: create(:setup_account))
+          split_account.splits << build(:split, percent: 50, extra_penny: false, parent_split_account: split_account, subaccount: create(:setup_account))
+          split_account.save
+        end
+      end
+
+      it "does not expire parent split_account" do
+        expect(split_account.expired?).to eq(false)
+      end
+    end
+
+    context "when any subaccount is expired" do
+      let(:expired_subaccount) { build(:setup_account, expires_at: Time.current) }
+
+      let(:split_account) do
+        build(:split_account, without_splits: true).tap do |split_account|
+          split_account.splits << build(:split, percent: 50, extra_penny: true, parent_split_account: split_account, subaccount: expired_subaccount)
+          split_account.splits << build(:split, percent: 50, extra_penny: false, parent_split_account: split_account, subaccount: create(:setup_account))
+          split_account.save
+        end
+      end
+
+      it "expires the parent split_account" do
+        expect(split_account.expired?).to eq(true)
+      end
+
+      context "when subaccount unexpires" do
+        before do
+          split_account
+          expired_subaccount.expires_at = Time.current + 1.year
+          expired_subaccount.save(validate: false)
+        end
+
+        it "unexpires the parent split_account" do
+          expect(split_account.reload.expired?).to eq(false)
```

Comment:
could this be `expect(split_account.reload).not_to be_expired` to be more meaningful?